### PR TITLE
Feature/stalagmite falls causing tiny aoe

### PR DESCRIPTION
**Pull request by Erick9Thor**
I need the @monikamnc work of crystal non regeneration to avoid regeneration of the fallen crystals.

To test just go to test escena and play. If you want add more stalagmites it's a prefab. Also there are a problem when i change the crystal parent, there is a frame where the scale of the crystal it's 50 I will need help with this.

### CODE REVIEW

**Review by Bernatmago — APPROVED** [review]
I lowered damage to 1, good job it seems to work fine! Maybe it could knock you out of the stalagmite as an improvement but it doesnt seem an issue for now.

**Review by baransrc — CHANGES_REQUESTED** [review]
Hey, thanks for the awesome work, as we talked on discord I'm requesting changes to block the pr from accidental merges as the requirements from the game design team changed. 

For other people reading this, now with the new document we reviewed on 26th of Sept, the stalagmites will be dealing damage when they fall, instead of creating explosive crystals on the landing point.

File: Gameplay/entities/crystals/Stalagmite.h [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+			Stalagmite(GameObject* game_object);
+			~Stalagmite() override = default;
+
+			StalagmiteState GetState() {
+				return stalagmite_state;
+			}
+
+			StalagmiteState GetPreviousState() {
+				return _previous_state;
+			}
```
Comment (baransrc) [inline Gameplay/entities/crystals/Stalagmite.h:31]:
```suggestion
			StalagmiteState GetState() 
			{
				return stalagmite_state;
			}

			StalagmiteState GetPreviousState() 
			{
				return _previous_state;
			}
```

File: Gameplay/entities/crystals/Stalagmite.h [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+			void SetPreviousState(StalagmiteState state)
+			{
+				_previous_state = state;
+			}
+
+			GameObject* GetGEO() {
+				return GEO;
+			}
```
Comment (baransrc) [inline Gameplay/entities/crystals/Stalagmite.h:45]:
```suggestion
			GameObject* GetGEO() 
			{
				return GEO;
			}
```

File: Gameplay/entities/crystals/Stalagmite.h [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+				return _explosion_effect;
+			}
+
+			void ActiveStalagmite();
+			void ActiveEffects();
+			void Falling();
+
+			bool IsStalagmiteCollapsed() {
+				return _stalagmite_collapsed;
+			}
```
Comment (baransrc) [inline Gameplay/entities/crystals/Stalagmite.h:58]:
```suggestion
			bool IsStalagmiteCollapsed() 
			{
				return _stalagmite_collapsed;
			}
```

File: Gameplay/misc/StalagmiteManager.cpp [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+void Hachiko::Scripting::StalagmiteManager::GenerateStalagmites()
+{
+	std::vector<GameObject*> stalagmites_children = game_object->children;
+
+	for (int i = 0; i < stalagmites_children.size(); i++)
+	{
+		Stalagmite* stalagmite = stalagmites_children[i]->GetComponent<Stalagmite>();
+		_stalagmites.push_back(stalagmite);
+	}
+}
```
Comment (baransrc) [inline Gameplay/misc/StalagmiteManager.cpp:37]:
```suggestion
void Hachiko::Scripting::StalagmiteManager::GenerateStalagmites()
{
    std::vector<GameObject*>& stalagmites_children = game_object->children;
    _stalagmites.clear();
    _stalagmites.reserve(stalagmites_children.size());

    for (GameObject* child : stalagmites_children)
    {
        Stalagmite* stalagmite = child->GetComponent<Stalagmite>();

        if (!stalagmite)
        {
            continue;
        }

        _stalagmites.push_back(stalagmite);
    }
}
```

File: Gameplay/misc/StalagmiteManager.cpp [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+					CombatManager::AttackStats attack_stats;
+					attack_stats.damage = 1;
+					attack_stats.knockback_distance = 0.0f;
+					attack_stats.width = 3.5f;
+					attack_stats.range = 4.5; // a bit bigger than its attack activation range
+					attack_stats.type = CombatManager::AttackType::CIRCLE;
+					_combat_manager->EnemyMeleeAttack(_stalagmites[i]->GetExplosionArea()
+						->GetTransform()->GetGlobalMatrix(), attack_stats);
+					
+
```
Comment (baransrc) [inline Gameplay/misc/StalagmiteManager.cpp:76]:
```suggestion
					
```

File: Gameplay/misc/StalagmiteManager.cpp [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+bool Hachiko::Scripting::StalagmiteManager::CheckPreviousStalagmite(int idx)
+{
+	if (idx == 0) // First stalagmite
+	{
+		return true;
+	}
+	else {
+		return _stalagmites[(int)idx - 1]->IsStalagmiteCollapsed();
+	}
```
Comment (baransrc) [inline Gameplay/misc/StalagmiteManager.cpp:135]:
```suggestion
    return (idx == 0) ? true : _stalagmites[static_cast<size_t>(idx - 1)]->IsStalagmiteCollapsed();
```

**Review by baransrc — APPROVED** [review]

**Pull request merged**